### PR TITLE
add adapter for landx project

### PR DESCRIPTION
**Pull request by KviatsetskiAndrey**
##### Name (to be shown on DefiLlama):
LandX Finance

##### Twitter Link:
https://twitter.com/landxfinance

##### List of audit links if any:
https://certificate.quantstamp.com/full/land-x-finance.pdf

##### Website Link:
https://landx.fi

##### Logo (High resolution, will be shown with rounded borders):
https://www.dropbox.com/scl/fi/gl7w1iueax84t8jhvmr0l/LNDX-200.png?rlkey=w5mw7g0t1c5vxilirqjsk1c1v&dl=0

##### Current TVL:
```
--- ethereum ---
USDC                      1.97 M
Total: 1.97 M

--- tvl ---
USDC                      1.97 M
Total: 1.97 M

------ TVL ------
ethereum                  1.97 M

total                    1.97 M
```

##### Treasury Addresses (if the protocol has treasury)
0x1A427D8717C16788a8A4a265804f489B9aB1e798

##### Chain:
Ethereum

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
landx-governance-token

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
N/A Pending Approval

##### Short Description (to be shown on DefiLlama):
Bridging the agriculture financing gap, LandX provides upfront capital to farmers in return for a percentage of their future harvests. The financing is secured by the underlying farmland, offering investors an opportunity to earn sustainable yield backed by tangible assets.

##### Token address and ticker if any:
0x08A1C30BBB26425c1031ee9E43FA0B9960742539
LNDX

##### Category (full list at https://defillama.com/categories) *Please choose only one:
https://defillama.com/protocols/RWA

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
Chainlink

##### forkedFrom (Does your project originate from another project):
N/A original code

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL is calculated as the total supply of xToken and cTokens mulitiplied by the market price with the addition of USDC locked in the RentFoundation contract.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/LandXit/land-x-smart-contracts

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/landx exports TVL: 
        
 
 
    ethereum                  1.97 M
    
    total                    1.97 M

**Comment by g1nt0ki** [comment]
@KviatsetskiAndrey thanks, I have refactored and merged it here, but got some doubt in the code: https://github.com/DefiLlama/DefiLlama-Adapters/pull/8493

File: projects/landx/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+async function getCTokensTVL(api) {
+  let sum = BigNumber.from(0)
+  for (const token of tokens) {
+    const price = BigNumber.from(await getCropPrice(token[0], api))
+    const cTokenSupply = BigNumber.from(await api.call({
+      abi: 'erc20:totalSupply',
+      target: token[1][1],
+    }))
+    const availableToClaim = BigNumber.from(await  getAvailableToClaim(token[1][0], api))
+    sum = sum.add(cTokenSupply.add(availableToClaim).mul(price).div(BigNumber.from(10 ** 9)))
```
Comment (g1nt0ki) [inline projects/landx/index.js:53]: why is the price here in 9 decimals and for xtoken 6 decimals.

So, when xToken is minted, the staked cToken is burnt? same question, you are counting the cTokens in availableToClaim, so, till claimed, these are not minted?

Comment (KviatsetskiAndrey) [reply projects/landx/index.js:53]: > why is the price here in 9 decimals and for xtoken 6 decimals.
CToken price equals to price of 1 kg of grain, and OraclePrices contains grain prices per mega tonne

>cTokens in availableToClaim, so, till claimed, these are not minted?
Yes, it is correct

>So, when xToken is minted, the staked cToken is burnt?
No, cTokens can't be staked, only xTokens. Staked xTokens generate yield in cTokens. CTokens can be burnt and the platform pays USDC for burnt cTokens

Comment (g1nt0ki) [reply projects/landx/index.js:53]: thanks for the answers @KviatsetskiAndrey

**Pull request closed without merging**